finagle-http/src/test/scala/com/twitter/finagle/http/ServerSideDecodingTest.scala

```diff
@@ -0,0 +1,94 @@
+package com.twitter.finagle.http
+
+import java.io.{ByteArrayOutputStream, OutputStream, PrintStream}
+import java.net.InetSocketAddress
+import java.util.zip.{DeflaterOutputStream, GZIPOutputStream}
+
+import com.twitter.finagle
+import com.twitter.finagle.Service
+import com.twitter.finagle.builder.ClientBuilder
+import com.twitter.io.Buf
+import com.twitter.util.{Await, Closable, Future}
+import org.jboss.netty.handler.codec.http.HttpHeaders
+import org.junit.runner.RunWith
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+import org.scalatest.{BeforeAndAfterAll, FunSuite}
+
+/** Provides tests for server side content decoding.
+  *
+  * If and when client side compression is implemented, this test should probably be removed in favour of a complete
+  * entry in [[EndToEndTest]]. Client side compression is currently made problematic by netty
+  * (see https://github.com/netty/netty/issues/4970).
+  */
+@RunWith(classOf[JUnitRunner])
+class ServerSideDecodingTest extends FunSuite with GeneratorDrivenPropertyChecks with BeforeAndAfterAll {
+  // Echo server (with decoding)
+  val server = finagle.Http.server
```

Comment:
feel free to ignore: I'd just spin up a new client and server for the test if I were you, and then we don't have to worry about spinning it down in an afterAll, we can just do it directly in the test.